pkg/registry/server/server.go

```diff
@@ -233,3 +239,125 @@ func ResolveLiteralConfig(resolver Resolver, resolverMetrics *metrics.Metrics) h
 		resolveAndRespond(resolver, unresolvedConfig, w, logger, resolverMetrics)
 	}
 }
+
+func ResolveAndMergeConfigsAndInjectTest(configs Getter, resolver Resolver, resolverMetrics *metrics.Metrics) http.HandlerFunc {
```

Comment:
`ResolveAndMergeConfigsAndInjectTest` needs a lot of URL query parameters:
```
/mergeConfigsWithInjectedTest?
    org=openshift,openshift&
    repo=rosa,hive&
    branch=master,master&
    variant=,periodic&
    injectTestFromOrg=openshift&
    injectTestFromRepo=release&
    injectTestFromBranch=master&
    injectTest=e2e-gcp-sdn-upgrade&
    injectTestFromVariant=ci-4.14
```
What about switching to a more compact form?
- `org/repo/branch/variant`:
```
/mergeConfigsWithInjectedTest?
    config=openshift/rosa/master,openshift/hive/master/periodic&
    test=openshift/release/master/ci-4.14
```

- `org/repo@branch_variant`
```
/mergeConfigsWithInjectedTest?
    config=openshift/rosa@master,openshift/hive@master_periodic&
    test=openshift/release@master_ci-4.14
```

Reply:
I agree that it is quite a few params, but I would rather stick with the existing pattern. These are actually the same params from `/configWithInjectedTest`. Since the vast majority of the time this will be called by `ci-operator` itself and the url is constructed programmatically, I don't think this is too bad.